### PR TITLE
feat: support tls alert key for project monitors

### PR DESCRIPTION
**Pull request by shahzad31**
Support tls alert key

```
heartbeat.monitors:
- type: http
  name: Todos Lightweight
  id: todos-lightweight
  enabled: true
  urls: "https://elastic.github.io/synthetics-demo/"
  schedule: '@every 3m'
  timeout: 16s
  alert.status.enabled: true
  alert.tls.enabled: true
```

### CODE REVIEW

**Comment by devcorpio** [comment]
LGTM!

--

To test this I also checked out the branch of the [Kibana's PR](https://github.com/elastic/kibana/pull/159697) locally.


1. Pushing a lightweight monitor setting `alert.status.enabled: true` and `alert.tls.enabled: true` in the .yaml file results in:



all_set_to_true


2. Pushing a lightweight monitor setting `alert.status.enabled: false` and `alert.tls.enabled: false` in the .yaml file results in:

all_set_to_false

3. Pushing a lightweight monitor without setting `alert.status.enabled` and `alert.tls.enabled` in the .yaml file results in:
Screenshot 2023-06-19 at 17 57 57

**Review by devcorpio — APPROVED** [review]

File: src/dsl/monitor.ts [hunk 1]
```diff
@@ -63,7 +63,10 @@ export type MonitorConfig = {
   params?: Params;
   playwrightOptions?: PlaywrightOptions;
   alert?: {
-    status: {
+    status?: {
+      enabled: boolean;
```
Comment (vigneshshanmugam) [inline src/dsl/monitor.ts:67]: nit: would type it as separate.

**Review by vigneshshanmugam — CHANGES_REQUESTED** [review]
There is support for configuring all of the monitors via global config `synthetics.config.ts`, is this intentional? 

Blocking this PR, before we agree on this as we have seen quite a few occasions lately that some options are not covered for all monitors.

**Comment by shahzad31** [comment]
> There is support for configuring all of the monitors via global config `synthetics.config.ts`, is this intentional?
> 
> Blocking this PR, before we agree on this as we have seen quite a few occasions lately that some options are not covered for all monitors.

Yes, it should be possible to configure this globally as well.

File: src/push/monitor.ts [hunk 2]
```diff
@@ -254,17 +258,36 @@ export function buildMonitorFromYaml(
...
+  }
+  if (config['alert.tls.enabled'] !== undefined) {
+    const value = config['alert.tls.enabled'];
+    delete config['alert.tls.enabled'];
+    alertConfig['tls'] = {
+      enabled: value,
+    };
+  }
+  const monitorAlertConfig =
+    Object.keys(alertConfig).length > 0 ? alertConfig : config.alert;
```
Comment (vigneshshanmugam) [inline src/push/monitor.ts:281]: not sure why this is required, wouldnt it be safer to just use do this
```
return {...globalConfig, ...alertConfig }
```

`config.alert` if present would fill in `alertConfig` anyways?

Comment (shahzad31) [reply src/push/monitor.ts]: done !!

**Review by vigneshshanmugam — APPROVED** [review]
LGTM pending one comment

**Pull request merged**